File: CodeChef/JuneChallenge2020/EvenMatrix/EVENM.py
import sys 

#below 2 lines are for running code on your local machine 
#Remove these 2 lines while using code online on codechef as they provide thier own input
sys.stdin = open("input.txt",'r')
sys.stdout = open("output.txt",'w')

# cook your dish here
t = int(input())
for j in range(t):
    n = int(input())
    ans=[]
    cnt=1
    ref=cnt
    for i in range(n):
        row=[]
        if(i%2==0):
            cnt=ref
            for k in range(n):
                row.append(cnt)
                cnt+=1
            ans.append(row)
            ref=cnt+n
        else:
            cnt=cnt+(n-1)
            for k in range(n):
                row.append(cnt)
                cnt-=1
            ans.append(row)
    
    for l in range(n):
        for m in range(n):
            print(ans[l][m],end=' ')
        print()
            

File: CodeChef/JuneChallenge2020/EvenMatrix/EVENM.cpp
#include <iostream>
using namespace std;
#include<vector>
int main() {
	// your code goes here
	//below 2 lines are for running code on your local machine 
    //Remove these 2 lines while using code online on codechef as they provide thier own input
    freopen("input.txt", "r", stdin);
    freopen("output.txt", "w", stdout);
    
	int t;
	cin>>t;
	for(int i=0;i<t;i++)
	{
	    int n;
	    cin>>n;
	    vector<vector <int>> ans(n);

	    int cnt=1;
	    int ref=cnt;
	    for(int j=0;j<n;j++)
	    {
	        //ans[j]=vector<int>(n);
	  
	        if(j%2==0)
	        {
	            cnt=ref;
    	        for(int k=0;k<n;k++)
    	        {
    	            ans[j].push_back(cnt++);
    	        }
    	        ref=cnt+n;
	        }
	        else
	        {
	            cnt=cnt+(n-1);
	            for(int k=0;k<n;k++)
    	        {
    	            ans[j].push_back(cnt--);
    	        } 
	        }
    	        //ans[j]=row;
    	    
	    }
	    for(int l=0;l<n;l++)
	    {
	        for(int m=0;m<n;m++)
	        {
	            cout<<ans[l][m]<<" ";
	        }
	        cout<<endl;
	    }
	}
	return 0;
}
